tests/common/mocks/KeyValueCache.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <gmock/gmock.h>
+
+#include <olp/core/cache/KeyValueCache.h>
+
+namespace olp {
+namespace testing {
+namespace mocks {
+
+class KeyValueCache : public olp::cache::KeyValueCache {
```

Comment:
```suggestion
class MockKeyValueCache : public olp::cache::KeyValueCache {
```
But the more urging question that is to be solved before: Why are we testing real behavior with mocks?

Reply:
`MockKeyValueCache `? Not `KeyValueCacheMock`?

Reply:
People may be interesting in measuring memory consumption with in-memory cache disabled.

Reply:
Which people? And gmock works without memory usage?
If you want to test without cache you set cache to nullptr but as this is not possible currently inside the OLP SDK you want to hide the unavoidable memory usage of the in-memory cache? I didn't recollect that we want to test OLP SDK without cache.